core/dbt/adapters/base/relation.py

```diff
@@ -254,8 +254,11 @@ def create_from(
                 )
             return cls.create_from_source(node, **kwargs)
         else:
+            # Can't use ManifestNode here because of parameterized generics
             if not isinstance(node, (ParsedNode)):
-                raise InternalException(f"type mismatch, expected ParsedNode but got {type(node)}")
+                raise InternalException(
+                    f"type mismatch, expected ManifestNode but got {type(node)}"
```

Comment:
They're actually the same thing. Mypy wouldn't let me use ManifestNode in an isinstance check. I'd rather not use ParsedNode in the message, because I think using the superclasses is more confusing...

Reply:
Maybe just a comment in the code to that effect then?

Reply:
In any case, it's really not a user-facing exception. It should only show up when a developer has made a mistake :)

Reply:
Sure, I just want to avoid spending 5 minutes wondering why these don't match next time I'm in there 😄

Reply:
The comment "# Can't use ManifestNode here because of parameterized generics" wasn't enough?